test/rotation_matrix_test.cpp

```diff
@@ -0,0 +1,250 @@
+#include "surface_perception/shape_extraction.h"
+#include "Eigen/Eigen"
+
+#include "pcl/ModelCoefficients.h"
+
+#include <gtest/gtest.h>
+namespace {
+// This test the equality of two matrix through approximation.
+bool isSameMatrix(const Eigen::Matrix3f& actual_matrix,
+		const Eigen::Matrix3f& expected_matrix) {
+  // Check matrix dimensions before testing content
+  if (actual_matrix.cols() != expected_matrix.cols()
+		  || actual_matrix.rows() != expected_matrix.rows()) {
+    return false;
+  }
+
+  return (actual_matrix - expected_matrix).array().abs().matrix().sum() < 0.0001;
+}
+
+void getHorizontalPlane(pcl::ModelCoefficients::Ptr model) {
```

Comment:
Return the models directly in the return value.

Reply:
The test helper functions removed.